### PR TITLE
FIX: Properly go through repeating groups values for all action type

### PR DESCRIPTION
**Pull request by laurennlam**
Only the delete method handled the case with repeating groups.
Now it's handled for all actions.

Fix [!17](https://github.com/KitwareMedical/dicom-anonymizer/issues/17)

### CODE REVIEW

File: dicomanonymizer/simpledicomanonymizer.py [hunk 1]
```diff
@@ -386,14 +378,27 @@ def anonymize_dataset(dataset: pydicom.Dataset, extra_anonymization_rules: dict
...
+        else:
+            action(dataset, tag)
+            try:
+                element = dataset.get(tag)
+            except:
+                print("Cannot get element from tag: ", tag_to_hex_strings(tag))
+
+            # Get private tag to restore it later
+            if element and element.tag.is_private:
+                private_tags.append(get_private_tag(dataset, tag))
```
Comment (finetjul) [inline dicomanonymizer/simpledicomanonymizer.py:401]: shouldn't you also do this in the "range" / "repeating-group" case ?

Comment (laurennlam) [reply dicomanonymizer/simpledicomanonymizer.py:401]: I didn't see any information about repeating group in private tag.
Moreover, the syntax we use int he project is proper to the project and should be encountered in private tags.

Comment (finetjul) [reply dicomanonymizer/simpledicomanonymizer.py:401]: > I didn't see any information about repeating group in private tag.

Fair enough, but did you see information that repeating groups can't exist in private tags ?

>  Moreover, the syntax we use int he project is proper to the project and should be encountered in private tags.

I'm not sure to understand what you mean here

Comment (laurennlam) [reply dicomanonymizer/simpledicomanonymizer.py:401]: It exists only two repeating groups which are not used in private tags : http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_7.6.html

Comment (finetjul) [reply dicomanonymizer/simpledicomanonymizer.py:401]: Understood.

This link is a good candidate to get into the commit message

Comment (laurennlam) [reply dicomanonymizer/simpledicomanonymizer.py:401]: Done !

**Comment by finetjul** [comment]
LGTM

**Pull request merged**